src/workerd/api/r2-bucket.c++

```diff
@@ -729,7 +750,20 @@ jsg::Promise<void> R2Bucket::delete_(jsg::Lock& js,
     const jsg::TypeHandler<jsg::Ref<R2Error>>& errorType) {
   return js.evalNow([&] {
     auto& context = IoContext::current();
-    auto client = context.getHttpClient(clientIndex, true, kj::none, "r2_delete"_kjc);
+    auto deleteKey = [&]() {
```

Comment:
it is weird to store lambda in the case. I suggest you call it immediately: 
```
kj::String deleteKey = [&](){ ... }();
```

(all because we don't have case expressions)